webdriver-extension/index.bs

```diff
@@ -0,0 +1,532 @@
+    text: GravitySensor; url: gravitysensor-interface
+urlPrefix: https://w3c.github.io/gyroscope; spec: GYROSCOPE
+  type: dfn
+    text: Gyroscope; url: gyroscope-interface
+urlPrefix: https://w3c.github.io/magnetometer; spec: MAGNETOMETER
+  type: dfn
+    text: Magnetometer; url: magnetometer-interface
+    text: UncalibratedMagnetometer; url: uncalibrated-magnetometer-interface
+urlPrefix: https://w3c.github.io/orientation-sensor; spec: ORIENTATION-SENSOR
+  type: dfn
+    text: AbsoluteOrientationSensor; url: absoluteorientationsensor-interface
+    text: RelativeOrientationSensor; url: relativeorientationsensor-interface
+urlPrefix: https://wicg.github.io/geolocation-sensor/; spec: GEOLOCATION-SENSOR
+  type: dfn
+    text: GeolocationSensor; url: geolocationsensor-interface
+urlPrefix: https://w3c.github.io/proximity; spec: PROXIMITY
+  type: dfn
+    text: ProximitySensor; url: proximity-sensor-interface
+</pre>
+
+Introduction {#intro}
+=====================
+
+Generic Sensor API [[GENERIC-SENSOR]] and its concrete Sensor APIs pose a challenge
+to test authors because to fully exercise their interfaces requires physical hardware
+devices that respond in predictable ways. To address this challenge this specification
+defines <a>extension commands</a> to the [[WebDriver]] specification for controlling
+[=mock sensor=] on the host that the user agent is running on.
+With these <a>extension commands</a> devices with particular properties can be created
+and their responses to requests are well defined.
+
+Mock Sensors {#mock-sensors}
+=====================
+
+A <dfn id="mock-sensor">mock sensor</dfn> simulates the behavior of a <a>platform sensor</a> in controlled ways.
+
+A [=mock sensor=] reports corresponding <dfn id="mock-sensor-reading">mock <a>sensor reading</a></dfn>, which is a source of
+mocking information about the environment, to the <a>Sensor</a> objects.
+
+Note: It's up to the user agent to expose what kinds of <a>mock sensor reading</a> to the <a>Sensor</a> objects.
+
+A [=mock sensor=] has an associated <a>latest reading</a> [=ordered map|map=], which holds the latest available [=mock sensor readings=].
+
+A [=mock sensor=] has an associated <a>requested sampling frequency</a> which is initial set to null.
+
+A [=mock sensor=] has an associated <a>sampling frequency</a> with supported bounds, the default values of
+supported bounds are freely defined by user agent.
+
+<div class='note'>
+Note: The [=mock sensor=] defined in this specification is not intended be used by non-testing-related web content.
+The UA MAY choose to expose [=mock sensor=] interface only when a runtime or compile-time flag has been set.
+</div>
+
+## MockSensorConfiguration dictionary ## {#dictionary-mocksensorconfiguration}
+
+<pre class='idl'>
+  dictionary MockSensorConfiguration {
+    required MockSensorType mockSensorType;
+    boolean sensorShouldConnected = true;
+    double? maxSamplingFrequency;
```

Comment:
Since these dictionary members are optional, also making them nullable ought to be redundant. The prose already says "If configuration.maxSamplingFrequency is present" and doesn't handle null, so that's good.